src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_SIZE = 10;
+    private static final double INCREMENT = 1.5;
+    private Object[] baseObjects;
+    private int size;
+
+    public ArrayList() {
+        baseObjects = new Object[START_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == baseObjects.length) {
+            resize();
+        }
+        baseObjects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index != 0) {
+            isInRange(index - 1, "Cannot use add for index ");
```

Comment:
let's not pass any messages to this method and use default message `Invalid index`